sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -226,6 +226,86 @@ class RocksDB(
     this
   }
 
+  /**
+   * Load from the start checkpoint version and apply all the changelog records to reach the
+   * end version. Note that this will copy all the necessary file from DFS to local disk as needed,
+   * and possibly restart the native RocksDB instance.
+   *
+   * @param startVersion
+   * @param endVersion
+   * @param readOnly
+   * @return
+   */
+  def load(startVersion: Long, endVersion: Long, readOnly: Boolean): RocksDB = {
+    assert(startVersion >= 0 && endVersion >= startVersion)
+    acquire(LoadStore)
+    recordedMetrics = None
+    logInfo(
+      log"Loading ${MDC(LogKeys.VERSION_NUM, endVersion)} from " +
+      log"${MDC(LogKeys.VERSION_NUM, startVersion)}")
+    try {
+      loadFromCheckpoint(startVersion, endVersion)
+
+      logInfo(
+        log"Loaded ${MDC(LogKeys.VERSION_NUM, endVersion)} from " +
+        log"${MDC(LogKeys.VERSION_NUM, startVersion)}")
+    } catch {
+      case t: Throwable =>
+        loadedVersion = -1  // invalidate loaded data
+        throw t
+    }
+    if (enableChangelogCheckpointing && !readOnly) {
+      // Make sure we don't leak resource.
```

Comment:
Hmm - would we need this block in this context ?

Reply:
I think this is another place I wanted it general. I will make a summary at the end.